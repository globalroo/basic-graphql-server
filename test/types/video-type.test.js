import { VideoType } from "../../src/types/video-type";
import { getRelatedVideos } from "../../src/data/data-api";

jest.mock("../../src/data/data-api", () => ({
	getRelatedVideos: jest.fn(),
}));

describe("GraphQL Types: VideoType", () => {
	test("VideoType returns has the correct structure", () => {
		expect (VideoType).toMatchSnapshot();
	});
	test("getRelatedVideos returns expected videos", async () => {
		const fields = VideoType._typeConfig.fields();
		fields.relatedVideos.resolve(undefined);
		expect(getRelatedVideos).toHaveBeenCalled();
	});
});
