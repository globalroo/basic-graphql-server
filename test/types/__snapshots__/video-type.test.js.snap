// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQL Types: VideoType VideoType returns has the correct structure 1`] = `"Video"`;
